eastdetector/__init__.py

```diff
@@ -153,39 +166,14 @@ class TextDetector:
     Adapted from: see top
     """
 
-    def __init__(self, checkpoint_path: str):
+    def __init__(self, pb_path: str):
         tf.logging.set_verbosity(tf.logging.ERROR)
 
-        self.__closed = False
-
-        checkpoint_state = tf.train.get_checkpoint_state(checkpoint_path)
-
-        if checkpoint_state is None:
-            raise RuntimeError("Failed to read the tensorflow checkpoint path from: {!r}".format(checkpoint_path))
+        self.pb_path = pb_path
 
-        model_path = os.path.join(checkpoint_path, os.path.basename(checkpoint_state.model_checkpoint_path))  # pylint: disable=no-member
+        self.layer_names = ["feature_fusion/Conv_7/Sigmoid", "feature_fusion/concat_3"]
 
-        self.__input_image = tf.placeholder(np.float32, shape=[None, None, None, 3], name='input_image')
-        self.__f_score, self.__f_geometry = eastdetector.model.model(self.__input_image, is_training=False)
 
-        self.__session = tf.Session()
-        global_step = tf.get_variable('global_step', [], initializer=tf.constant_initializer(0), trainable=False)
-        variable_averages = tf.train.ExponentialMovingAverage(0.997, global_step)
-        saver = tf.train.Saver(variable_averages.variables_to_restore())
-        saver.restore(self.__session, model_path)
-
-    def __enter__(self):
```

Comment:
Just FYI: deleting these functions is changing the interface, so you will have to bum the major version

More info about versioning:
https://semver.org/